SETUP/smoketests/pageload-smoketest.sh

```diff
@@ -0,0 +1,253 @@
+    'quiz/tuts/tut_p_old_2.php'
+    'quiz/tuts/tut_p_old_3.php'
+    'quiz/tuts/tut_p_thorn.php'
+)
+
+TEAMS_URLS=(
+    'stats/members/jointeam.php?tid=44' # NB performs an action
+    'stats/members/quitteam.php?tid=44' # NB performs an action
+    'stats/teams/team_topic.php?team=44' # Adds a forum post!
+    'stats/members/mbr_list.php?tid=44'
+    'stats/members/mbr_xml.php?username=teststeel'
+    'stats/members/mdetail.php?id=1'
+    'stats/teams/new_team.php'
+    'stats/teams/tdetail.php?tid=44'
+    'stats/teams/teams_xml.php?tid=44'
+    'stats/teams/tedit.php?tid=44'
+    'stats/teams/tlist.php'
+)
+STATS_URLS=(
+    'stats/PP_unknown.php'
+    'stats/default.php'
+    'stats/equilibria.php'
+    'stats/misc_stats1.php?tally_name=F1&start=2024-01&end=2024-02'
+    'stats/misc_user_graphs.php'
+    'stats/pages_in_states.php'
+    'stats/pages_proofed_graphs.php?tally_name=P3' # Slow to evaluate!
+    'stats/percent_users_who_proof.php'
+    'stats/pm_stats.php'
+    'stats/pp_stage_goal.php'
+    'stats/pp_stats.php'
+    'stats/ppv_stats.php'
+    'stats/projects_Xed_graphs.php?which=posted'
+    'stats/proof_stats.php?tally_name=F2'
+    'stats/release_queue.php'
+    'stats/requested_books.php'
+    'stats/round_backlog.php'
+    'stats/round_backlog_days.php'
+    'stats/stats_central.php'
+    'stats/user_logon_graphs.php'
+)
+STYLES_URLS=(
+    'styles/design_philosophy.php'
+    'styles/style_demo.php'
+)
+BASE_TOOL_URLS=(
+    'tools/change_sr_commitment.php?projectid=projectID5e23a810ef693&action=commit'
+)
+
+URLS=(
+    ${NOLOGIN_URLS[*]}
+    ${BASE_URLS[*]}
+    ${API_URLS[*]}
+    ${FAQ_URLS[*]}
+    ${MISC_URLS[*]}
+    ${QUIZ_URLS[*]}
+    ${TEAMS_URLS[*]}
+    ${STATS_URLS[*]}
+    ${STYLES_URLS[*]}
+    ${BASE_TOOL_URLS[*]}
+)
+
+# TODO Add remaining ~95 pages
+# TODO Add accounts/logout.php
+
+function do_exit()
+{
+    if [ $ERR != 0 ]; then
+        echo Exiting with error
+        echo Server log:
+        cat $SERVER_LOG
+    fi
+    # Kill all children of this script (ie the php -S, if it's still alive)
+    /bin/kill -INT $(jobs -p)
+}
+
+trap do_exit EXIT
+
+# HACK: Give php -S time to start up
+sleep .5
+
+# Log in using the login page; and capture the session ID from the cookies
+# in the response header
+if [ $CACHE_COOKIE = 1 -a -f .smoketest-cookie ]; then
+    COOKIE=$(cat .smoketest-cookie)
+else
+    COOKIE=$(curl http://$IP_PORT/accounts/login.php \
+        -X POST \
+        --data "userNM=$DP_USER&userPW=$DP_PASS" \
+        --output /dev/null \
+        --silent \
+        --dump-header - | \
+        sed -n 's/Set-Cookie: \(DPT_Session=[^;]*\);.*$/\1/p')
+    [ $CACHE_COOKIE = 1 ] && echo $COOKIE > .smoketest-cookie
+fi
+
+for URL in ${URLS[@]}; do
+    echo -n > $SERVER_LOG
+    echo Getting $URL
+    curl --silent \
+        --cookie "$COOKIE" \
+        --header "X-API-KEY: $API_KEY" \
+        http://$IP_PORT/$URL > $OUT
+    if grep -qE 'PHP (Notice|Warning|Fatal error)' $SERVER_LOG; then
```

Comment:
Is there a way we can test this check? For example, have a known bad page? That way if PHP updates and changes their error messaging format in some way, this check would fail?

Reply:
Yup. I've done exactly that by hand. I'll add it as a pre-check to the test.

Reply:
Done in the python rewrite in the `server_ready` function